src/engine/plugins/table.cpp

```diff
@@ -98,14 +98,20 @@ Status TablePlugin::HandleRequest(const RoutingAlgorithmsInterface &algorithms,
     std::vector<api::TableAPI::TableCellRef> estimated_pairs;
 
     // Scan table for null results - if any exist, replace with distance estimates
-    if (params.fallback_speed != INVALID_FALLBACK_SPEED || params.scale_factor != 1)
+    if (params.fallback_speed != INVALID_FALLBACK_SPEED || params.scale_factor != 1 ||
+        (params.min_stoppage_penalty != INVALID_MINIMUM_STOPPAGE_PENALTY &&
+         params.max_stoppage_penalty != INVALID_MAXIMUM_STOPPAGE_PENALTY))
     {
         for (std::size_t row = 0; row < num_sources; row++)
         {
             for (std::size_t column = 0; column < num_destinations; column++)
             {
                 const auto &table_index = row * num_destinations + column;
                 BOOST_ASSERT(table_index < result_tables_pair.first.size());
+                // Zero out the diagonal
+                if (result_tables_pair.first[table_index] != MAXIMAL_EDGE_DURATION && row == column)
```

Comment:
because each waypoint in the source/targets list is unique we ended up adding the penalty to all of them, which means even the diagonal of the matrix will have non-0 entries if the penalty was enabled. this is kind of messy to reset the diagonal here :frowning_face: